### PR TITLE
Extend the fail2ban instructions with a hint on how to make X-Real-IP…

### PR DESCRIPTION
**Pull request by dosera**
Following the merging of #14959 - Gitea is a lot more strict regarding the interpretation of `X-Real-IP` and `X-Forwarded-For` headers.

This PR updates the fail2ban documentation to include hints to set: `REVERSE_PROXY_TRUSTED_PROXIES` and `REVERSE_PROXY_LIMIT` appropriately.

See discussion in #16443

### CODE REVIEW

File: docs/content/doc/usage/fail2ban-setup.en-us.md [hunk 1]
```diff
@@ -108,3 +108,12 @@ this to your Nginx configuration so that IPs don't show up as 127.0.0.1:
...
 proxy_set_header X-Real-IP $remote_addr;
 ```
+
+The security options in `app.ini` need to be adjusted to allow the interpretation of the headers
+as well as the list of IP addresses and networks that describe trusted proxy servers
+(find more information in the corresponding [PR](https://github.com/go-gitea/gitea/pull/14959)).
+
+```
+REVERSE_PROXY_LIMIT = 1
+REVERSE_PROXY_TRUSTED_PROXIES = *
```
Comment (techknowlogick) [inline docs/content/doc/usage/fail2ban-setup.en-us.md]: Perhaps the default for this in docs be `127.0.0.1/8`

Comment (dosera) [reply docs/content/doc/usage/fail2ban-setup.en-us.md]: of course

**Review by techknowlogick — APPROVED** [review]

File: docs/content/doc/usage/fail2ban-setup.en-us.md [hunk 2]
```diff
@@ -108,3 +108,12 @@ this to your Nginx configuration so that IPs don't show up as 127.0.0.1:
 ```
 proxy_set_header X-Real-IP $remote_addr;
 ```
+
+The security options in `app.ini` need to be adjusted to allow the interpretation of the headers
+as well as the list of IP addresses and networks that describe trusted proxy servers
+(find more information in the corresponding [PR](https://github.com/go-gitea/gitea/pull/14959)).
```
Comment (lafriks) [inline docs/content/doc/usage/fail2ban-setup.en-us.md]: It would be better to add link to docs somewhere instead of PR

Comment (zeripath) [reply docs/content/doc/usage/fail2ban-setup.en-us.md]:
```suggestion
(See the [configuration cheat sheet](https://docs.gitea.io/en-us/config-cheat-sheet/#security-security) for more information).
```

**Review by zeripath — APPROVED** [review]

**Pull request merged**